app/core/Tracker2/BaseTracker.js

```diff
@@ -1,3 +1,22 @@
+export const TRACKER_LOGGING_ENABLED_QUERY_PARAM = 'tracker_logging';
+
+export const DEFAULT_USER_TRAITS_TO_REPORT = [
+  'email', 'anonymous', 'dateCreated', 'hourOfCode', 'name', 'referrer', 'testGroupNumber', 'testGroupNumberUS',
+  'gender', 'lastLevel', 'siteref', 'ageRange', 'schoolName', 'coursePrepaidID', 'role', 'firstName', 'lastName',
+  'dateCreated'
+]
+
+export function extractDefaultUserTraits(me) {
+  return DEFAULT_USER_TRAITS_TO_REPORT.reduce((obj, key) => {
+    const meAttr = me[key]
+    if (typeof meAttr !== 'undefined' && meAttr !== null) {
```

Comment:
What is the difference between the `typeof` check and `meAttr !== undefined`?

Reply:
I'm not sure, this is just how I've always done it.  I'll have to take a look at the difference.